wordstats/cognate_info.py

```diff
@@ -0,0 +1,391 @@
+import codecs
+import os
+from datetime import datetime
+
+from python_translators.translators.glosbe_translator import Translator
+from python_translators.translators.glosbe_over_tor_translator import GlosbeOverTorTranslator
+from python_translators.translation_query import TranslationQuery
+
+from sqlalchemy import Table
+import configparser
+
+from wordstats.loading_from_hermit import load_language_from_hermit
+
+from .word_info import WordInfo, UnknownWordInfo
+from .utils.mem_footprint import total_size
+from .base_service import BaseService, Base
+from .config import MAX_WORDS, SEPARATOR_PRIMARY, SEPARATOR_SECONDARY
+from .metrics_computers import *
+from .cognate_files_path import *
+from .cognate_db import *
+from .getchunix import read_single_keypress
+from .edit_distance_function_factory import WordDistanceFactory
+from collections import defaultdict
+
+from time import sleep
+
+
+class CognateInfo(object):
+    """
+
+        Responsibilities of this class:
+        - compute the candidates
+        - maintain the blacklist / whitelist (add_to_whitelist / add_to_blacklist / save_evaluation )
+        - load /save blacklist / whitelist / candidates from file / DB to memory
+
+
+
+    """
+
+    def __init__(self, primary, secondary, distance_computer_class: WordDistanceFactory, author:str = ""):
+        """
+
+            either load from file, or compute if needed
+
+        :param primary:
+        :param secondary:
+        :param method_name:
+        """
+        self.primary = primary
+        self.secondary = secondary
+        self.whitelist = defaultdict(list)
+        self.candidates = defaultdict(list)
+        self.blacklist = defaultdict(list)
+        self.distance_computer = distance_computer_class(primary, secondary, author)
+        self.author = author
+
+    def best_guess(self):
+        best_dict = dict(self.candidates)
+        for key in self.blacklist.keys():
+            best_dict.pop(key)
+
+        return best_dict
+
+    # generates candidates based on distance function func and word lists
+    # def apply_distance_metric(self, wordlist1, wordlist2, func):
+
+    def compute(self):
+
+        wordlist1 = list(load_language_from_hermit(self.primary).word_info_dict.keys())
+        wordlist2 = list(load_language_from_hermit(self.secondary).word_info_dict.keys())
+
+        self.candidates = defaultdict(list)
+        i = 0
+        for w1 in wordlist1:
+            for w2 in wordlist2:
+                if self.distance_computer.is_candidate(w1, w2):
+                    self.candidates[w1].append(w2)
+
+
+    def compute_translator(self, translator:Translator, save:Boolean = False):
```

Comment:
this is not a self-explanatory name.
think of a better name, and also add docstring.